main/network.c

```diff
@@ -859,54 +859,52 @@ php_socket_t php_network_connect_socket_to_host(const char *host, unsigned short
 		if (sa) {
 			/* make a connection attempt */
 
-			if (bindto) {
-				struct sockaddr *local_address = NULL;
-				int local_address_len = 0;
+			while (bindto) {
+				struct {
+					int len;
+					union {
+						struct sockaddr common;
+						struct sockaddr_in in4;
+#if HAVE_IPV6 && HAVE_INET_PTON
+						struct sockaddr_in6 in6;
+#endif
+					} u;
```

Comment:
You can drop the `u` here and use an anonymous union.

Reply:
Forgot we are in C99